Clean up selecting guard logic

marklawlor opened the pull request:
## Proposed Change:

Clean up the `selecting` logic to be handled by interactjs.

Inspired by https://github.com/BHP-DevHub/react-timeline-9000/pull/206#issuecomment-764418044.
 
## Change type

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

lilfolr reviewed (changes requested):
Tried this locally and it seems to break the selecting.
Can you confirm if it works for you? Haven't had time to check why it isnt working

marklawlor commented:
> Tried this locally and it seems to break the selecting.
> Can you confirm if it works for you? Haven't had time to check why it isnt working

Selecting is working for us, but that maybe because we have also changed the event listeners on the row (eg #206 & other changes not pushed yet). Looking at our patches, we have removed the row events completely - which might be why it works for us. I'm guessing the `e.preventDefault()` in `_handleItemRowEvent` should be after the `if (this.selecting) {}` guard, as the `onMouseOver` and `onMouseLeave` events will be cancelled and never bubble to `interactjs`

We're in the crunch towards a deadline, but I'll come back and have a look at this afterwards. These PRs are me just trying to push some of our changes upstream.

lilfolr commented:
Awesome thanks - really appreciate you taking the time to do so
